### PR TITLE
Add GDPR support

### PR DESCRIPTION
**Pull request by Eseperio**
I have added basic support for GDPR regulation.
It contains:
- Enable by configuration
- Checkbox for data consent in registration form
- Register of consent date and consent in db
- Privacy settings page
- Data portability (export personal data)
- Right to be forgotten (deletion of personal data)
- Tests
- Docs (configuration and helpful guide)

> Changelog not updated until resolution of this PR

### CODE REVIEW

**Comment by maxxer** [comment]
Wow! Awesome, thanks! I'll try to give a look at it in the next days

File: tests/_data/schema.sql [hunk 1]
```diff
@@ -81,4 +84,113 @@ CREATE TABLE `user` (
   UNIQUE KEY `user_unique_username` (`username`)
 ) ENGINE=InnoDB AUTO_INCREMENT=7 DEFAULT CHARSET=utf8;
 
+
+DROP TABLE IF EXISTS `auth_assignment`;
```
Comment (maxxer) [inline tests/_data/schema.sql]: why you added RBAC tables to the schema? usuario uses yii's core rbac tables and migration

Comment (Eseperio) [reply tests/_data/schema.sql]: To use `$I->amLoggedInAs()`

**Comment by tonydspaniard** [comment]
@Eseperio Thanks for such amount of work and support for this module. Looks like a great addition.

File: tests/_data/schema.sql (hunk 1 above)
Comment (tonydspaniard) [reply tests/_data/schema.sql]: @Eseperio it means that Usuario already uses RBAC`s tables from Yii.

Comment (Eseperio) [reply tests/_data/schema.sql]: This is the schema only for tests, not migrations.

**Comment by maxxer** [comment]
I'm ok with merging  this PR after few minor issues have been addressed. @tonydspaniard do you have time for a release after that?

**Comment by maxxer** [comment]
I've merged your contribution, thank you very much!!

**Comment by Eseperio** [comment]
> An additional great feature would be to force users who haven't accepted privacy policy yet to be forced to do. Consider the case when you add GDPR support to an existing site, and you want existing users to see and accept the document

I would like to discuss how we should do this. 
There are many options
1. Add a flash on the app until the user gives consent
2. Prevent access to any action and redirect to consent page.

**Comment by maxxer** [comment]
Please move the discussion to #197

**Comment by Eseperio** [comment]
Moved!

**Pull request merged**